src/SFML/Graphics/Font.cpp

```diff
@@ -837,8 +839,9 @@ bool Font::setCurrentSize(unsigned int characterSize) const
 Font::Page::Page(bool smooth) : nextRow(3)
 {
     // Make sure that the texture is initialized by default
-    sf::Image image;
-    image.create({128, 128}, Color(255, 255, 255, 0));
+    std::optional<Image> maybeImage = Image::create({128, 128}, Color(255, 255, 255, 0));
+    assert(maybeImage.has_value());
```

Comment:
With the current implementation of `Image::create()` this is safe, because the only way for it to return `std::nullopt` is for the dimensions to be zero, which is obviously not the case here. But if someone adds another failure mode later (i.e., allocation failure), then this code can now invoke undefined behavior in Release mode (where `assert(...)` is stripped off). This feels brittle.

In my opinion, `std::optional` should be treated seriously; if it's declared as optional, then that means there's a possibility for it to fail, and you must check if it has an appropriate value even if you think you don't need to. Or at the very least, use `.value()` and let it throw/terminate if you make a wrong assumption. 

If you think this is nonsensical here, I would argue this is because `Image::create()` shouldn't return `std::optional` to signal a pre-condition violation (see my other comment on the implementation of this function). If you take that out, then `Image::create()` returns a plain `Image`, and you don't need to check for anything. But then if someone decides that allocation failure should be caught and cause `std::nullopt`, then you can no longer assume that `Image::create()` will succeed, and you must check the optional again.